Tribler/community/tunnel/conversion.py

```diff
@@ -58,13 +85,13 @@ def _decode_cell(self, placeholder, offset, data):
 
     def _encode_create(self, message):
         payload = message.payload
-        packet = pack("!IHH20s", payload.circuit_id, len(payload.node_public_key),
-                      len(payload.key), payload.node_id) + payload.node_public_key + payload.key
+        packet = pack("!IHH20s?", payload.circuit_id, len(payload.node_public_key),
+                      len(payload.key), payload.node_id, payload.exit_candidates) + payload.node_public_key + payload.key
```

Comment:
The problem is the number maximum of 4 possible candidates returned by the last hop. When the next hop becomes the endpoint/exit-node, n-1 should at least return 4 candidates with at least 1 exit-node-enabled candidate. Chances are small, because most users prefer not to be exit-node. That's why the flag is added here.